qutebrowser/misc/sessions.py

```diff
@@ -466,13 +467,16 @@ def session_load(self, name, clear=False, temp=False, force=False):
             temp: Don't set the current session for :session-save.
             force: Force loading internal sessions (starting with an
                    underline).
+            delete: Delete the saved session once it has loaded.
         """
         if name.startswith('_') and not force:
             raise cmdexc.CommandError("{} is an internal session, use --force "
                                       "to load anyways.".format(name))
         old_windows = list(objreg.window_registry.values())
         try:
             self.load(name, temp=temp)
+            if delete:
+                self.session_delete(name)
```

Comment:
The `session_delete` method is mainly intended as command handler, not for use in code - you might want to use `self.delete` instead, as you do handle `SessionError` and others yourself.